src/intermine_nlp/template.clj

```diff
@@ -0,0 +1,86 @@
+(ns intermine-nlp.template
+  (:require [imcljs.fetch :as im-fetch]
+            [imcljs.path :as im-path]
+            [clojure.edn :as edn]
+            [clojure.java.io :as io]
+            [clojure.pprint :refer [pprint]]
+            [clojure.core.match :refer [match]]
+            [clojure.walk :refer [walk]]
+            [intermine-nlp.util :as util]))
+
+(defn load-local-templates
+  "Load a local copy of templates EDN file (backup for fetch-templates).
+  Options are:
+  medic|fly|fly-b|human|yeast|rat|mouse"
+  [db-name]
+  (let [path (str "test/queries/" db-name "_templates.edn")]
+    (util/read-edn path)))
+
+
+(defn fetch-templates-
+  "Fetch templates for the given dataset. Defaults to flymine.
+  Options are:
+  medic|fly|fly-b|human|yeast|rat|mouse
+  Accepts either a database name (fly, yeast...) or a service"
+  [db]
+  (match [db]
+         [{:root root}]
+         (try
+           (im-fetch/templates {:root root})
+           (catch java.lang.Exception e (print "Couldn't access database")))
+         [db-name]
+         (let [url (case db-name
+                     "medic"    "medicmine.jcvi.org/medicmine"
+                     "fly"      "www.flymine.org/query"
+                     "human"    "www.humanmine.org/humanmine"
+                     "yeast"    "yeastmine.yeastgenome.org/yeastmine"
+                     "rat"      "ratmine.mcw.edu/ratmine"
+                     "mouse"    "www.mousemine.org/mousemine"
+                     "fly-beta" "beta.flymine.org/beta"
+                     "www.flymine.org/query")
```

Comment:
Falling back to fly as a default is error-prone. I just tried this ```(def bean-templates (template/fetch-templates- "beanmine"))``` and it looked successful, but actually I had fly templates and might not have realised it. probably better to assert that the db-name isn't known rather than returning results for the wrong one. I think similar logic is present in other places, too.